src/Client/Web/Bit.Client.Web.BlazorUI/Components/Pickers/BitDatePicker.razor.cs

```diff
@@ -159,7 +160,7 @@ public async Task HandleDateChoose(int dayOfWeek, int day, int month)
                 IsOpen = false;
                 BitDate date = new(currentYear,month,day,dayOfWeek);
                 selectedDate = OnSelectDate is not null ? OnSelectDate.Invoke(date): GetSelectedDateString(date);
-                if (ValueChanged.HasDelegate is true)
+                if (ValueChanged.HasDelegate is true && ValueHasBeenSet)
```

Comment:
this condition and its location are not correct, plz follow other implementations in the project.